scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/http/SchedulerBindings.java

```diff
@@ -40,10 +31,6 @@ public record SchedulerBindings(
     ScheduleAction scheduleAction,
     GenerateSchedulingLibAction generateSchedulingLibAction
 ) implements Plugin {
-  public SchedulerBindings {
-    Objects.requireNonNull(schedulerService, "schedulerService must be non-null");
-    Objects.requireNonNull(scheduleAction, "scheduleAction must be non-null");
-  }
```

Comment:
why remove this requirement?

(I may have written this comment twice, I can't remember. My bad if you get this twice)

Reply:
Adding `requireNonNull()` guards to `record` classes kind of depends on personal style, but since this PR should not attempt to make "2nd order refactoring changes" (refactoring on top of refactoring scheduler server) I'll revert this change.

In general I think these `requireNonNull()`s on `record` definitions mostly add clutter.

Reply:
Reverted this change. Although I did take out the optional `"schedulerService must be non-null"` messages since they are redundant when these `requireNonNull` asserts fail.

Reply:
Do records require non null automatically?

Reply:
They don't. So I should really say it's really a matter of preference, not style.

Reply:
This is one area I kind of sway back & forth on. I think @Twisol also felt a bit ambivalent here too

Reply:
Yeah, I really wish records automatically had non-null checks. I'd like to say I add them depending on how far the nulls could potentially propagate... but usually it's a matter of "do I really want to have all this boilerplate?".

I'd say non-null checks are ideal, but not worth it for certain kinds of records. This one is the type that would fail instantly on attempting to interact with the service, so we don't necessarily need to crash out much sooner than that.